Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm

```diff
@@ -662,14 +662,30 @@ static void registerLogHook()
     os_log_set_hook(OS_LOG_TYPE_DEFAULT, ^(os_log_type_t type, os_log_message_t msg) {
         if (msg->buffer_sz > 1024)
             return;
-        char* messageString = os_log_copy_message_string(msg);
-        String logString = String::fromUTF8(messageString);
-        free(messageString);
 
+        String logFormat = String::fromUTF8(msg->format);
         String logChannel = String::fromUTF8(msg->subsystem);
         String logCategory = String::fromUTF8(msg->category);
 
-        callOnMainRunLoop([logChannel = logChannel.isolatedCopy(), logCategory = logCategory.isolatedCopy(), logString = logString.isolatedCopy(), type] {
+        Vector<uint8_t> buffer(msg->buffer, msg->buffer_sz);
+        Vector<uint8_t> privdata(msg->privdata, msg->privdata_sz);
+
+        static NeverDestroyed<Ref<WorkQueue>> queue(WorkQueue::create("Log Queue",  WorkQueue::QOS::Background));
+
+        dispatch_async(queue.get()->dispatchQueue(), [logFormat = logFormat.isolatedCopy(), logChannel = logChannel.isolatedCopy(), logCategory = logCategory.isolatedCopy(), type = type, buffer = WTFMove(buffer), privdata = WTFMove(privdata)] {
```

Comment:
Note for the future, you would want WTFMove(string).isolatedCopy() for all of these. Otherwise, you're going to end up doing a deep copy unnecessarily.

However, given that I suggested using `CString` earlier, you just need to WTFMove() now, no isolatedCopy().